### ISSUE TITLE
Feature Request: Integration with OpenVPN and PIA?

### ISSUE THREAD

**Issue by CaptainSolo2** [linuxserver/docker-rutorrent#69]
Im just going to toss this out there as a question/ idea...

After installing this container on my NAS, Im happy to say it's working and with your help I was able to resolve the closed Listen port issue. But once i enable the VPN profile on the NAS (to connect to PIA), the listen port closes again. Clearly, there is something going on with the VPN that is blocking the Listen Port, but so far I have been unable to figure out how to resolve this. (Port forwarding on the NAS is probably possible, but I havent found a clear solution in their Help or Forums yet.)

So... the question/idea is this.... There are a quite a few Docker containers for a Deluge/PIA setup on GitHub - Have you guys considered adding a feature to this container that would do the same thing? I think it would be incredible if you did. You may even be able to reference the Deluge/PIA configs on Git to help you do it.

So this is just a thought... but it be nice enhancement to support an OpenVPN PIA connection as part of the container (similar to whats being done on the Deluge/PIA containers). However, I dont know what would be required on the development end to do that or if it's even something that you may be interested in doing.

**Comment by WellingR** [comment]
Not completely sure what you mean with the listen port issue. But do you mean you cannot access rutorrent while connected through the vpn?

Here is what I do (note: I omit environment variables in the examples below, see documentation on the specific dockers to see how to configure)

Start your vpn docker and assign it a name `--name=pia`

```
docker run -d --cap-add=NET_ADMIN --name=pia itsdaspecialk/pia-openvpn
```

start docker-rutorrent using the network of the pia container `--net=container:pia` do not expose any ports (this is not possible when using --net=container). Again note the name of the container `--name=rutorrent`

```
docker run -d --net=container:pia --name=rutorrent linuxserver/rutorrent
```

To be able to access the http ports fire up an nginx proxy which is linked to the `rutorrent` container in the `pia` container network (`--link pia:rutorrent`)


```
docker run -it --name nginx -p 80:80 -p 443:443 --restart=always \
            --link pia:rutorrent \
            -d dperson/nginx \
            -w "http://rutorrent:80/;/rutorrent"
```

When all containers are running you should be able to access rutorrent using localhost:80/rutorrent

More details on the nginx proxy can be found in the readme of https://github.com/dperson/openvpn-client

**Comment by CaptainSolo2** [comment]
Thanks for the response... but that's not quite the issue. I did a more detailed write-up on the link below - the diagram may make it simpler to explain.
https://forum.synology.com/enu/viewtopic.php?f=145&t=137192

The short version is, when I have the Docker container set up, its exposing my IP address, so I connected it to a VPN (PIA, in this case), but for some reason the Listen Port is closed, which means I am "passively" seeding and not "actively" seeding (People who are actively seeding can see both types - Passive seeders can only connect to other Passive seeders - I think.)

I am pretty sure I know what's causing the problem, but have no idea how to port forward on the IP address given by the VPN to the Listen port needed by rTorrent within Docker.

**Comment by WellingR** [comment]
Understood. However I have no experience with this (however I am interested too).

As far as I know PIA has some port forwarding scripts which allow you to request a port to forward. This script return the number of the port that will be forwarded. Perhaps it would be possible to write this number to the config file before starting the rutorrent docker?

I am not sure if you actually need to configure port-forwarding on your NAS at all since the port that is forwarded is a port on the VPN host, so not a port on your network. In the end you are running a VPN to hide your ip, you would not want to leak it by exposing some information about your private network.

**Comment by CaptainSolo2** [comment]
I have read elsewhere that PIA has port forwarding scripts, but dont know how to include that in this without potentially breaking something that I dont have the skills to fix. (Command line changes are pretty intimidating to someone who isnt very knowledgeable with those.)

I've spent untold hours trying to fix this on my own, to no avail. I am pretty sure there is something I can do through the NAS UI that would correct this - some magic combination of port settings - but Ive tried 100 different ways and none of them worked. There is very likely a command line solution, but it would take someone with more skills than me to find it, then create a write-up so that us mere mortals could implement it.

I see several Docker/Deluge containers that include a PIA config option out of the box (so to speak), hence the feature request here. If it were implemented, I would be happy to do a complete write up with an install video - I just cant figure out the sequential set up of steps to correct this.
